File: task-03/java/Sub-Task4/code4.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class DiamondPattern {

    public static void main(String[] args) {
        try {
            // Read the number from input.txt
            File inputFile = new File("input.txt");
            Scanner scanner = new Scanner(inputFile);
            int n = scanner.nextInt();
            scanner.close();

            // Generate the diamond pattern and write it to output.txt
            FileWriter writer = new FileWriter("output.txt");
            generateDiamond(n, writer);
            writer.close();

            System.out.println("Diamond pattern generated in output.txt");
        } catch (IOException e) {
            System.out.println("An error occurred while processing the file.");
            e.printStackTrace();
        }
    }

    public static void generateDiamond(int n, FileWriter writer) throws IOException {
        // Upper part of the diamond
        for (int i = 1; i <= n; i++) {
            writeLine(writer, n - i, 2 * i - 1);
        }

        // Lower part of the diamond
        for (int i = n - 1; i >= 1; i--) {
            writeLine(writer, n - i, 2 * i - 1);
        }
    }

    private static void writeLine(FileWriter writer, int spaces, int stars) throws IOException {
        // Write spaces
        for (int j = 0; j < spaces; j++) {
            writer.write(' ');
        }

        // Write stars
        for (int j = 0; j < stars; j++) {
            writer.write('*');
        }

        // Move to the next line
        writer.write('\n');
    }
}

File: task-03/python/Sub-Task4/code4.py
def generate_diamond(n):
    diamond = []

    # Upper part of the diamond
    for i in range(1, n + 1):
        spaces = ' ' * (n - i)
        stars = '*' * (2 * i - 1)
        diamond.append(spaces + stars)

    # Lower part of the diamond
    for i in range(n - 1, 0, -1):
        spaces = ' ' * (n - i)
        stars = '*' * (2 * i - 1)
        diamond.append(spaces + stars)

    return diamond

def write_to_file(diamond, filename='output.txt'):
    with open(filename, 'w') as f:
        f.write('\n'.join(diamond) + '\n')

def main():
    # Read the number from input.txt
    with open('input.txt', 'r') as f:
        n = int(f.read().strip())

    # Generate the diamond pattern
    diamond = generate_diamond(n)

    # Write the diamond pattern to output.txt
    write_to_file(diamond)

    print("Diamond pattern generated in output.txt")

if __name__ == "__main__":
    main()

